dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -114,6 +114,7 @@ void onCancelPermissionRequest(String permission) {
    */
   void requestPermission(String permission) {
     int permissionCode = getPermissionCodeForPermission(permission);
+    this.firstTimeRequested = !ActivityCompat.shouldShowRequestPermissionRationale(activity, permission);
```

Comment:
Didn't you have any comments about how the interface was being used? Something related to add a parameter to the listener signature instead of add another method to the interface?